app/frontend/components/text_size_settings.js

```diff
@@ -1,31 +1,79 @@
 class TextSizeSettings {
   constructor() {
-    this._settingsObject = window.ipcSettings;
+    this.defaultTextSize = 10; // in em*10 so not to deal with float
+    this.minSize = 7;
+    this.maxSize = 20;
+    this._textSizeValue = this.defaultTextSize;
+    this._settingsKey = 'verse-text-size';
     this.openMenuButton = '.text-size-settings';
     this.menuContainer = '.text-size-menu';
     this.menuIsOpened = false;
+    this.stylesheet = null;  // CSSStyleSheet
   }
 
-  init() {
-    $(this.openMenuButton).bind('click', (event) => {
-      event.stopPropagation()
-      if(this.menuIsOpened) {
+  async init() {
+    if (this.stylesheet !== null) {
+      return;
+    }
+
+    var styleEl = $('<style id="dynamic-text-size" />')
+    $("head").append(styleEl);
+    this.stylesheet = styleEl[0].sheet;
+
+    $('#app-container').find(this.openMenuButton).bind('click', (e) => {
+      e.stopPropagation()
+      if (this.menuIsOpened) {
         this.hideTextSizeMenu();
       } else {
         this.showTextSizeMenu();
       }
     });
+
+    $('#app-container').find('.text-size-reset').bind('click', () => {
+      this._textSizeValue = this.defaultTextSize;
+      this.updateStyle();
+      this.saveConfig();
+    });
+
+    $('#app-container').find('.text-size-decrease').bind('click', () => {
+      if(this._textSizeValue <= this.minSize+1) {
+        return;
+      }
+      this._textSizeValue -= 1;
+      this.updateStyle();
+      this.saveConfig();
+    });
+
+    $('#app-container').find('.text-size-increase').bind('click', () => {
+      if(this._textSizeValue >= this.maxSize-1) {
+        return;
+      }
+      this._textSizeValue += 1;
+      this.updateStyle();
+      this.saveConfig();  
+    });
+
+    if (window.ipcSettings) {
+      this._textSizeValue = await window.ipcSettings.get(this._settingsKey, this.defaultTextSize);
+    }
+
+    this.updateStyle();
   }
 
   showTextSizeMenu() {
-    console.log(`--- show ${this.menuContainer}`);
     var currentVerseListMenu = app_controller.getCurrentVerseListMenu();
-    var $menuButton = currentVerseListMenu.find('.display-options-button');
+    var $menuButton = currentVerseListMenu.find(this.openMenuButton);
     $menuButton.addClass('ui-state-active');
+    var buttonOffset = $menuButton.offset();
     var $menuContainer = currentVerseListMenu.find(this.menuContainer);
 
-    var top_offset = $menuButton.top + $menuButton.height() + 1;
-    var left_offset = $menuButton.left + $menuButton.width()/2 - $menuContainer.width()/2;    
+    var top_offset = buttonOffset.top + $menuButton.height() + 2;
+    var left_offset = buttonOffset.left - 2;
```

Comment:
Code style: Let's try to be consistent using camel case for all variables. There is some legacy code where this is not the case, but let's try to use this style for all new code.